### ISSUE TITLE
Compatibility with xtcetools GUI

### ISSUE THREAD

**Issue by gmarchetx** [yamcs/pymdb#2]
Dear developer,

I think we should be able to use the very convenient GUI xtcetools available [here](https://gitlab.com/dovereem/xtcetools), but it cannot open XTCE files that are generated by pymdb - neither can it open the example MDB in the YAMCS quickstart. I know this is more of an issue for xtcetools, and I did raise this as an issue (on a Github fork first), but it seems neither have been maintained for years so I do not have any hope. 

One thing that does relate to this project though, is that the files from YAMCS quickstart and generated by Pymdb seem different from xtcetools's built-in examples, starting from the fact they do not have xtce: in their tags. I'm not sure which is correct.

If you want to test it yourself, I apt installed openjdk-21-jdk & maven, git cloned master then ran mvn package -DskipTests & java -jar xtcetools/target/XXXX_with_dependencies.jar in the local repo. See issue #32 [here](https://gitlab.com/dovereem/xtcetools/-/issues/32).

Many thanks

**Comment by xpromache** [comment]
Having or not xtce: in the tags is a matter of namespace declaration at the beginning of the file. 

if you replace
xmlns="http://www.omg.org/spec/XTCE/20180204"
with 
xmlns:xtce="http://www.omg.org/spec/XTCE/20180204"

then you can use xtce: prefix in front of all xtce tags.


About xtcetools: did you try with java 8 or 11?

**Comment by fqqb** [comment]
I could confirm the following error displayed in XTCETools:

```
XML Schema Compliance Fatal: ( line 4 column 116 ) unexpected element (uri:"http://www.omg.org/spec/XTCE/20180204", local:"SpaceSystem"). Expected elements are <{http://www.omg.org/space/xtce}SpaceSystem>, ...
```

You can work around this error by manually modifying the default XML namespace in the root element from `http://www.omg.org/spec/XTCE/20180204` to `http://www.omg.org/space/xtce`.

I then ran into more issues, and it turns out that the XSD that XTCETools uses for validation is not actually XTCE v1.2, but maybe some earlier draft (filename mentions the year 2014). See: https://gitlab.com/dovereem/xtcetools/-/blob/master/src/main/resources/org/omg/space/xtce/schema/SpaceSystemV1.2-27Feb2014-mods.xsd#L29

The actual OMG v1.2 is available from the following link, and should correspond with what PyMDB is doing: https://www.omg.org/spec/XTCE/20180204/SpaceSystem.xsd

So in summary, it seems like an issue to XTCETools. My guess is that XTCETools was primarily designed for XTCE v1.0 and v1.1, and that the v1.2 support was never fully implemented.

**Comment by gmarchetx** [comment]
Thanks to both. @fqqb yes you are correct, I am in touch with the developers that were kind enough to answer, and I can confirm XTCE 1.2 was never fully implemented. I will discuss this with the developers and see what can be done, in the meantime may I recommend to keep this issue open as, unless I am mistaken, there is no other GUI alternative for YAMCS mission databases based on XTCE and so helping out with xtcetools may be beneficial to YAMCS users. For example, in our case, we are currently using Excel databases but git versioning is challenging with them, and XTCE files are, I must say, awful for users compared to Excel workbooks.

P.S: @xpromache I have tested JDK 8, 11, 21, and one of the xtcetools developers also reproduced with 17. It seems like the exception thrown may be a misnomer for a format-related issue.
